pxl/compress.py

```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
+from PIL import Image
+from resizeimage import resizeimage
+
+# TODO: These values are picked at random wihtout much thought
+WIDTHS = [1600, 800, 400, 250]
+
+
+def compress_image(local_filename: Path) -> [Path]:
+    """Compresses the image to different sizes and saves them to /tmp, returns a list of paths"""
+
+    image_paths = []
+    with open(str(local_filename)) as file:
+        with Image.open(local_filename) as image:
+            # Copy the original image to /tmp
+            path = Path("/tmp").joinpath(local_filename.name)
+            image.save(path)
+            image_paths.append(path)
+
+            # Get the original dimensions
+            real_w, real_h = image.size
+            for w in WIDTHS:
+                # Prevent upscaling
+                if w >= real_w:
+                    continue
+
+                # Scale the image and save it to the /tmp directory
+                scaled = resizeimage.resize_width(image, w)
+                scaled_w, scaled_h = scaled.size
+                scaled_path = Path(f"/tmp/{local_filename.name}.scaled{scaled_w}x{scaled_h}")
```

Comment:
Can we also use [`tempfile.TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) instead of writing to `/tmp` directly? (Less crossplatform tweaks in case we ever want to port this)

Reply:
Great idea